includes/wikia/api/LogEventsApiController.class.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * API which allows you to add items to Special:Log
+ *
+ * This is basically HTTP interface to the these PHP methods:
+ *
+ * LogPage::__construct ($type, $rc=true, $udp= 'skipUDP')
+ * LogPage::addEntry ($action, $target, $comment, $params=array(), $doer=null)
+ *
+ * @see https://www.mediawiki.org/wiki/Manual:LogPage.php
+ * @see https://www.mediawiki.org/wiki/Manual:Log_actions
+ */
+
+class LogEventsApiController extends WikiaApiController {
+
+	/**
+	 * Add a new entry to logging table
+	 *
+	 * @throws BadRequestException
+	 */
+	public function add() {
+		global $wgTheSchwartzSecretToken, $wgTitle, $wgUser;
+
+		if ( !$this->request->wasPosted() ) {
+			throw new BadRequestException( 'This request must be POSTed' );
+		}
+
+		if ( !Wikia\Security\Utils::matchToken( $wgTheSchwartzSecretToken, $this->request->getVal( 'token' ) ) ) {
+			throw new BadRequestException( 'This request must provide a valid token' );
+		}
+
+		$type = $this->request->getVal( 'type' );
+		$action = $this->request->getVal( 'action' );
+		$comment = $this->request->getVal( 'comment' );
+		$params = json_decode( $this->request->getVal( 'params', '[]' ) );
+
+		try {
+			$entry = new LogPage( $type );
+			$id = $entry->addEntry( $action, $wgTitle, $comment, $params, $wgUser );
+		}
+		catch ( DBQueryError $e ) {
+			// 1048 Column 'log_type' cannot be null
+			throw new BadRequestException( $e->getMessage() );
```

Comment:
Exposing exact error message doesn't necessarily need to be secure. Can we improve the reporting here (log the error and return general message with request id so we could correlate reports later)?